tests/channel_test.c

```diff
@@ -281,8 +280,10 @@ static int s_test_channel_refcount(struct aws_allocator *allocator, void *ctx) {
 
     /* Release hold 2/2. The handler and channel should be destroyed. */
     aws_channel_release_hold(channel);
+    ASSERT_SUCCESS(aws_mutex_lock(&destroy_mutex));
     ASSERT_SUCCESS(aws_condition_variable_wait(&destroy_condition_variable, &destroy_mutex));
     ASSERT_TRUE(aws_atomic_load_int(&destroy_called));
+    ASSERT_SUCCESS(aws_mutex_unlock(&destroy_mutex));
```

Comment:
outside the scope of fixing immediate tsan errors, but:

Should we replace all uses of `aws_condition_variable_wait()` with `aws_condition_variable_wait_pred()`? I would even vote for **completely removing the vanilla wait() call from aws-c-common.** These tests are only using vanilla wait() because it required less code to get working than wait_pred(). But vanilla wait() can spuriously wake up and break the assumptions of the test.

From the [pthread docs](https://linux.die.net/man/3/pthread_cond_wait):
> When using condition variables there is always a Boolean predicate involving shared variables associated with each condition wait that is true if the thread should proceed. Spurious wakeups from the pthread_cond_timedwait() or pthread_cond_wait() functions may occur. Since the return from pthread_cond_timedwait() or pthread_cond_wait() does not imply anything about the value of this predicate, the predicate should be re-evaluated upon such return.

Reply:
I don’t necessarily want to remove it, but we should definitely not be using it in tests. This particular test was written prior to us adding the predicate variant.

Reply:
Yes, replace these calls with their `_pred` equivalents in the tests. Also, document this behavior for future users: https://github.com/awslabs/aws-c-common/issues/580